### DISCUSSION TITLE
How to disable Spell Checking (autocorect) in a toga.TextInput for Android platform ?

### DISCUSSION THREAD

**Discussion by MarusGradinaru** [beeware/toga#2801] [General]
I made an Adroid app with with BeeWare Toga. In it I have a toga.TextInput widget. And when I write someting it underline some words with a red line. I gues this is autocorect... Can I disable somehow this behavior ? My app targets only Android platform.

**Comment by rmartin16** [comment]
This is one strategy that worked for me:
```python
text_input = toga.TextInput()

if toga.platform.current_platform == "android":
    from android.text import InputType
    text_input._impl.native.setInputType(InputType.TYPE_TEXT_FLAG_NO_SUGGESTIONS)
```

**Reply by freakboy3742** [reply]
Spelling autocorrect is a common enough feature of text inputs that we should probably expose a control for this in the public API. It might end up being a no-op on some platforms, but for platforms where it *is* controllable, it would be a nice improvement.

**Reply by freakboy3742** [reply]
Logged as #2805.

**Reply by MarusGradinaru** [reply]
Thank you so much ! I traied that too, but without "_impl", that's why was not working...

**Reply by MarusGradinaru** [reply]
By the way, is there some workaround to set the TextInput value without triggering any of the "on_...." events ? 
When my app starts, it update the preferences pannel (which include TextInput's) with the default validated values. In this case I don't want to trigger any events, I just need to update the default value of the TextInput...

**Reply by rmartin16** [reply]
Only way I see is to set the desired default value when you create the `TextInput` or wait to assign handlers until the value is what you want.
